issues with sundials 3.0.0

current develop can't compile with sundials 3.0.0. Some interfaces must have changed, numerous build errors:
```
     1297  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In member function ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&)’:
  >> 1298  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:427:25: error: ‘struct ARKodeMemRec’ has no member named ‘ark_setupNonNull’
     1299               ARKode_mem->ark_setupNonNull = true;
     1300                           ^
  >> 1301  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:444:25: error: ‘struct ARKodeMemRec’ has no member named ‘ark_MassSetupNonNull’
     1302               ARKode_mem->ark_MassSetupNonNull = true;
     1303                           ^
     1304  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In instantiation of ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&) [with VectorType = dealii::Vector<double>]’:
     1305  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:468:18:   required from here
  >> 1306  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:423:32: error: no matches converting function ‘t_arkode_solve_jacobian’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector, N_Vector, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*, struct _generic_N_Vector*, struct _generic_N_Vector*)’}
     1307           ARKode_mem->ark_lsolve = t_arkode_solve_jacobian<VectorType>;
     1308                                  ^
     1309  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:138:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_jacobian(ARKodeMem, N_Vector, N_Vector, N_Vector, N_Vector)
     1310       int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
     1311           ^
  >> 1312  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:439:32: error: no matches converting function ‘t_arkode_solve_mass’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*)’}
     1313           ARKode_mem->ark_msolve = t_arkode_solve_mass<VectorType>;
     1314                                  ^
     1315  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:188:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_mass(ARKodeMem, N_Vector, N_Vector)
     1316       int t_arkode_solve_mass(ARKodeMem arkode_mem,
     1317           ^
     1318  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In instantiation of ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&) [with VectorType = dealii::BlockVector<double>]’:
     1319  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:469:18:   required from here
  >> 1320  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:423:32: error: no matches converting function ‘t_arkode_solve_jacobian’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector, N_Vector, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*, struct _generic_N_Vector*, struct _generic_N_Vector*)’}
     1321           ARKode_mem->ark_lsolve = t_arkode_solve_jacobian<VectorType>;
     1322                                  ^
     1323  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:138:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_jacobian(ARKodeMem, N_Vector, N_Vector, N_Vector, N_Vector)
     1324       int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
     1325           ^
  >> 1326  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:439:32: error: no matches converting function ‘t_arkode_solve_mass’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*)’}
     1327           ARKode_mem->ark_msolve = t_arkode_solve_mass<VectorType>;
     1328                                  ^
     1329  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:188:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_mass(ARKodeMem, N_Vector, N_Vector)
     1330       int t_arkode_solve_mass(ARKodeMem arkode_mem,
     1331           ^
     1332  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In instantiation of ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&) [with VectorType = dealii::TrilinosWrappers::MPI::Vector]’:
     1333  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:474:18:   required from here
  >> 1334  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:423:32: error: no matches converting function ‘t_arkode_solve_jacobian’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector, N_Vector, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*, struct _generic_N_Vector*, struct _generic_N_Vector*)’}
     1335           ARKode_mem->ark_lsolve = t_arkode_solve_jacobian<VectorType>;
     1336                                  ^
     1337  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:138:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_jacobian(ARKodeMem, N_Vector, N_Vector, N_Vector, N_Vector)
     1338       int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
     1339           ^
  >> 1340  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:439:32: error: no matches converting function ‘t_arkode_solve_mass’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*)’}
     1341           ARKode_mem->ark_msolve = t_arkode_solve_mass<VectorType>;
     1342                                  ^
     1343  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:188:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_mass(ARKodeMem, N_Vector, N_Vector)
     1344       int t_arkode_solve_mass(ARKodeMem arkode_mem,
     1345           ^
     1346  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In instantiation of ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&) [with VectorType = dealii::TrilinosWrappers::MPI::BlockVector]’:
     1347  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:475:18:   required from here
  >> 1348  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:423:32: error: no matches converting function ‘t_arkode_solve_jacobian’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector, N_Vector, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*, struct _generic_N_Vector*, struct _generic_N_Vector*)’}
     1349           ARKode_mem->ark_lsolve = t_arkode_solve_jacobian<VectorType>;
     1350                                  ^
     1351  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:138:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_jacobian(ARKodeMem, N_Vector, N_Vector, N_Vector, N_Vector)
     1352       int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
     1353           ^
  >> 1354  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:439:32: error: no matches converting function ‘t_arkode_solve_mass’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*)’}
     1355           ARKode_mem->ark_msolve = t_arkode_solve_mass<VectorType>;
     1356                                  ^
     1357  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:188:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_mass(ARKodeMem, N_Vector, N_Vector)
     1358       int t_arkode_solve_mass(ARKodeMem arkode_mem,
     1359           ^
     1360  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In instantiation of ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&) [with VectorType = dealii::PETScWrappers::MPI::Vector]’:
     1361  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:480:18:   required from here
  >> 1362  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:423:32: error: no matches converting function ‘t_arkode_solve_jacobian’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector, N_Vector, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*, struct _generic_N_Vector*, struct _generic_N_Vector*)’}
     1363           ARKode_mem->ark_lsolve = t_arkode_solve_jacobian<VectorType>;
     1364                                  ^
     1365  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:138:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_jacobian(ARKodeMem, N_Vector, N_Vector, N_Vector, N_Vector)
     1366       int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
     1367           ^
  >> 1368  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:439:32: error: no matches converting function ‘t_arkode_solve_mass’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*)’}
     1369           ARKode_mem->ark_msolve = t_arkode_solve_mass<VectorType>;
     1370                                  ^
     1371  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:188:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_mass(ARKodeMem, N_Vector, N_Vector)
     1372       int t_arkode_solve_mass(ARKodeMem arkode_mem,
     1373           ^
     1374  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc: In instantiation of ‘void dealii::SUNDIALS::ARKode<VectorType>::reset(const double&, const double&, const VectorType&) [with VectorType = dealii::PETScWrappers::MPI::BlockVector]’:
     1375  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:481:18:   required from here
  >> 1376  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:423:32: error: no matches converting function ‘t_arkode_solve_jacobian’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector, N_Vector, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*, struct _generic_N_Vector*, struct _generic_N_Vector*)’}
     1377           ARKode_mem->ark_lsolve = t_arkode_solve_jacobian<VectorType>;
     1378                                  ^
     1379  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:138:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_jacobian(ARKodeMem, N_Vector, N_Vector, N_Vector, N_Vector)
     1380       int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
     1381           ^
  >> 1382  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:439:32: error: no matches converting function ‘t_arkode_solve_mass’ to type ‘int (*)(struct ARKodeMemRec*, N_Vector) {aka int (*)(struct ARKodeMemRec*, struct _generic_N_Vector*)’}
     1383           ARKode_mem->ark_msolve = t_arkode_solve_mass<VectorType>;
     1384                                  ^
     1385  /home/denis/spack/var/spack/stage/dealii-develop-bm27meuslaphwji6bj7o5fpwbnb2bsph/dealii/source/sundials/arkode.cc:188:9: note: candidate is: template<class VectorType> int dealii::SUNDIALS::{anonymous}::t_arkode_solve_mass(ARKodeMem, N_Vector, N_Vector)
     1386       int t_arkode_solve_mass(ARKodeMem arkode_mem,
     1387           ^
  >> 1388  source/sundials/CMakeFiles/obj_sundials_release.dir/build.make:65: recipe for target 'source/sundials/CMakeFiles/obj_sundials_release.dir/arkode.cc.o' failed
  >> 1389  make[2]: *** [source/sundials/CMakeFiles/obj_sundials_release.dir/arkode.cc.o] Error 1
     1390  make[2]: Leaving directory '/tmp/denis/spack-stage/spack-stage-Cc2o43/dealii/spack-build'
  >> 1391  CMakeFiles/Makefile2:5379: recipe for target 'source/sundials/CMakeFiles/obj_sundials_release.dir/all' failed
  >> 1392  make[1]: *** [source/sundials/CMakeFiles/obj_sundials_release.dir/all] Error 2
```

--- a/doc/news/changes/minor/20171103DanielArndt
+++ b/doc/news/changes/minor/20171103DanielArndt
@@ -1,0 +1,3 @@
+Improved: Both Sundials-2.7.0 and Sundials-3.0.0 are supported.
+<br>
+(Daniel Arndt, 2017/11/03)

--- a/include/deal.II/base/config.h.in
+++ b/include/deal.II/base/config.h.in
@@ -216,6 +216,31 @@
     (major)*1000000 + (minor)*10000 + (subminor)*100 + (patch))
 #endif
 
+ /*
+  * SUNDIALS:
+  *
+  *  Note: The following definitions will be set in sundials_config.h,
+  *        so we don't repeat them here.
+  *
+  *  SUNDIALS_VERSION_MAJOR
+  *  SUNDIALS_VERSION_MINOR
+  *  SUNDIALS_VERSION_PATCH
+  */
+
+ #define DEAL_II_SUNDIALS_VERSION_GTE(major,minor,subminor) \
+   ((SUNDIALS_VERSION_MAJOR * 10000 + \
+     SUNDIALS_VERSION_MINOR * 100 + \
+     SUNDIALS_VERSION_SUBMINOR) \
+     >=  \
+     (major)*10000 + (minor)*100 + (subminor))
+
+ #define DEAL_II_SUNDIALS_VERSION_LT(major,minor,subminor) \
+   ((SUNDIALS_VERSION_MAJOR * 10000 + \
+     SUNDIALS_VERSION_MINOR * 100 + \
+     SUNDIALS_VERSION_SUBMINOR) \
+     <  \
+     (major)*10000 + (minor)*100 + (subminor))
+
 /*
  * PETSc:
  *

--- a/include/deal.II/sundials/ida.h
+++ b/include/deal.II/sundials/ida.h
@@ -35,9 +35,12 @@
 
 #include <ida/ida.h>
 #include <ida/ida_spils.h>
-#include <ida/ida_spgmr.h>
-#include <ida/ida_spbcgs.h>
-#include <ida/ida_sptfqmr.h>
+#include <sundials/sundials_config.h>
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
+#  include <ida/ida_spgmr.h>
+#  include <ida/ida_spbcgs.h>
+#  include <ida/ida_sptfqmr.h>
+#endif
 #include <nvector/nvector_serial.h>
 #include <sundials/sundials_math.h>
 #include <sundials/sundials_types.h>

--- a/source/sundials/arkode.cc
+++ b/source/sundials/arkode.cc
@@ -32,6 +32,8 @@
 #include <deal.II/base/utilities.h>
 #include <deal.II/sundials/copy.h>
 
+#include <sundials/sundials_config.h>
+
 #include <iostream>
 #include <iomanip>
 #include <arkode/arkode_impl.h>
@@ -137,7 +139,9 @@ namespace SUNDIALS
     template<typename VectorType>
     int t_arkode_solve_jacobian(ARKodeMem arkode_mem,
                                 N_Vector b,
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
                                 N_Vector,
+#endif
                                 N_Vector ycur,
                                 N_Vector fcur)
     {
@@ -186,8 +190,13 @@ namespace SUNDIALS
 
     template<typename VectorType>
     int t_arkode_solve_mass(ARKodeMem arkode_mem,
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
                             N_Vector b,
-                            N_Vector)
+                            N_Vector
+#else
+                            N_Vector b
+#endif
+                           )
     {
       ARKode<VectorType> &solver = *static_cast<ARKode<VectorType> *>(arkode_mem->ark_user_data);
       GrowingVectorMemory<VectorType> mem;
@@ -424,7 +433,9 @@ namespace SUNDIALS
         if (setup_jacobian)
           {
             ARKode_mem->ark_lsetup = t_arkode_setup_jacobian<VectorType>;
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
             ARKode_mem->ark_setupNonNull = true;
+#endif
           }
       }
     else
@@ -441,7 +452,9 @@ namespace SUNDIALS
         if (setup_mass)
           {
             ARKode_mem->ark_msetup = t_arkode_setup_mass<VectorType>;
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
             ARKode_mem->ark_MassSetupNonNull = true;
+#endif
           }
       }
 

--- a/source/sundials/ida.cc
+++ b/source/sundials/ida.cc
@@ -387,7 +387,9 @@ namespace SUNDIALS
 
     IDA_mem->ida_lsetup = t_dae_lsetup<VectorType>;
     IDA_mem->ida_lsolve = t_dae_solve<VectorType>;
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
     IDA_mem->ida_setupNonNull = true;
+#endif
 
     status = IDASetMaxOrd(ida_mem, data.maximum_order);
     AssertIDA(status);

--- a/source/sundials/kinsol.cc
+++ b/source/sundials/kinsol.cc
@@ -22,17 +22,24 @@
 #include <deal.II/base/utilities.h>
 #include <deal.II/lac/block_vector.h>
 #ifdef DEAL_II_WITH_TRILINOS
-#include <deal.II/lac/trilinos_parallel_block_vector.h>
-#include <deal.II/lac/trilinos_vector.h>
+#  include <deal.II/lac/trilinos_parallel_block_vector.h>
+#  include <deal.II/lac/trilinos_vector.h>
 #endif
 #ifdef DEAL_II_WITH_PETSC
-#include <deal.II/lac/petsc_parallel_block_vector.h>
-#include <deal.II/lac/petsc_parallel_vector.h>
+#  include <deal.II/lac/petsc_parallel_block_vector.h>
+#  include <deal.II/lac/petsc_parallel_vector.h>
 #endif
 #include <deal.II/base/utilities.h>
 #include <deal.II/sundials/copy.h>
 
-#include <kinsol/kinsol_dense.h>
+#include <sundials/sundials_config.h>
+#if DEAL_II_SUNDIALS_VERSION_GTE(3,0,0)
+#  include <sunmatrix/sunmatrix_dense.h>
+#  include <sunlinsol/sunlinsol_dense.h>
+#  include <kinsol/kinsol_direct.h>
+#else
+#  include <kinsol/kinsol_dense.h>
+#endif
 
 #include <iostream>
 #include <iomanip>
@@ -254,12 +261,20 @@ namespace SUNDIALS
         if (setup_jacobian)
           {
             KIN_mem->kin_lsetup = t_kinsol_setup_jacobian<VectorType>;
+#if DEAL_II_SUNDIALS_VERSION_LT(3,0,0)
             KIN_mem->kin_setupNonNull = true;
+#endif
           }
       }
     else
       {
+#if DEAL_II_SUNDIALS_VERSION_GTE(3,0,0)
+        const auto J = SUNDenseMatrix(system_size, system_size);
+        const auto LS = SUNDenseLinearSolver(u_scale, J);
+        status = KINDlsSetLinearSolver(kinsol_mem, LS, J);
+#else
         status = KINDense(kinsol_mem, system_size);
+#endif
         AssertKINSOL(status);
       }
 

